From 1c88fe3db7d3b478beb54362fe983bde28dd10a0 Mon Sep 17 00:00:00 2001
From: Hill <hillyu@live.com>
Date: Thu, 27 Aug 2020 20:12:19 +0800
Subject: [PATCH] fixed losefulscreen

---
 dwm.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/dwm.c b/dwm.c
index 90869d2..fd3cc97 100644
--- a/dwm.c
+++ b/dwm.c
@@ -553,8 +553,9 @@ buttonpress(XEvent *e)
 	click = ClkRootWin;
 	/* focus monitor if necessary */
 	if ((m = wintomon(ev->window)) && m != selmon) {
-		unfocus(selmon->sel, 1);
+        Client *oldc =selmon->sel;
 		selmon = m;
+		unfocus(oldc, 1);
 		focus(NULL);
 	}
 	if (ev->window == selmon->barwin) {
@@ -972,8 +973,9 @@ enternotify(XEvent *e)
 	c = wintoclient(ev->window);
 	m = c ? c->mon : wintomon(ev->window);
 	if (m != selmon) {
-		unfocus(selmon->sel, 1);
+        Client *oldc =selmon->sel;
 		selmon = m;
+		unfocus(oldc, 1);
 	} else if (!c || c == selmon->sel)
 		return;
 	focus(c);
@@ -1033,8 +1035,9 @@ focusmon(const Arg *arg)
 		return;
 	if ((m = dirtomon(arg->i)) == selmon)
 		return;
-	unfocus(selmon->sel, 0);
+    Client *oldc =selmon->sel;
 	selmon = m;
+	unfocus(oldc, 1);
 	focus(NULL);
 	warp(selmon->sel);
 }
@@ -1368,8 +1371,9 @@ motionnotify(XEvent *e)
 	if (ev->window != root)
 		return;
 	if ((m = recttomon(ev->x_root, ev->y_root, 1, 1)) != mon && mon) {
-		unfocus(selmon->sel, 1);
+        Client *oldc =selmon->sel;
 		selmon = m;
+		unfocus(oldc, 1);
 		focus(NULL);
 	}
 	mon = m;
@@ -2216,6 +2220,8 @@ unfocus(Client *c, int setfocus)
 {
 	if (!c)
 		return;
+	if (c->isfullscreen && ISVISIBLE(c) && c->mon == selmon)
+		setfullscreen(c, 0);
 	grabbuttons(c, 0);
 	XSetWindowBorder(dpy, c->win, scheme[SchemeNorm][ColBorder].pixel);
 	if (setfocus) {
-- 
2.28.0

